### PR TITLE
Update to Factorio 1.1

### PR DESCRIPTION
**Pull request by skaupper**
I made the necessary changes to be compliant with the newest Factorio version.

Relevant changelogs: [here](https://forums.factorio.com/viewtopic.php?f=3&t=91657) and [here](https://forums.factorio.com/viewtopic.php?f=3&t=91858).

Testing locally with editing an old mod version worked (the save loaded properly and nothing broke obviously).

Feel free to suggest further changes.

Cheers!

### CODE REVIEW

File: prototypes/prototypes.lua [hunk 1]
```diff
@@ -11,8 +11,8 @@ data:extend(
         type = 'shortcut',
         name = 'yarm-selector',
         order = "a[yarm]",
-        action = 'create-blueprint-item',
-        item_to_create = 'yarm-selector-tool',
+        action = 'spawn-item',
+        item_to_spawn = 'yarm-selector-tool',
```
Comment (narc0tiq) [inline prototypes/prototypes.lua:15]: Doesn't `yarm-selector-tool` not also need the `spawnable` flag?

File: resmon.lua [hunk 2]
```diff
@@ -2,6 +2,7 @@ require "util"
 require "libs/array_pair"
 require "libs/ore_tracker"
 require "mod-gui"
```
Comment (narc0tiq) [inline resmon.lua:4]: Remove this extra `require`.

**Comment by narc0tiq** [comment]
Assuming it all works, aside from the aforementioned nitpicks, I'll merge it in with many thanks.

File: prototypes/prototypes.lua (hunk 1 above)
Comment (ahmedcharles) [reply prototypes/prototypes.lua:15]: Yes, it does. This is sufficient for the game to run, but the selection tool doesn't spawn the item on the cursor if the flag isn't set.

**Comment by ahmedcharles** [comment]
The map tags now have icons that look like pins (the new default) rather than having the icon set to the ore type. They still seem to work just fine, but it's a regression in terms of graphics. I also didn't investigate how to fix it.

**Comment by narc0tiq** [comment]
They never used the ore type as the chart tag icon (at least, not in a
release build - I tried it in dev once but it looked pretty bad). What it
should have is the ore entity or item (don't remember which) in rich text
in the tag text.

**Comment by skaupper** [comment]
@narc0tiq I resolved your comments and also bumped the version number to 0.8.103.

The map pins still look fine (in my opinion). Even if they can be improved I think that should be resolved in a separate PR, shouldn't it?

File: prototypes/prototypes.lua [hunk 3]
```diff
@@ -11,8 +11,9 @@ data:extend(
...
+        spawnable = true,
```
Comment (narc0tiq) [inline prototypes/prototypes.lua:16]: The `shortcut` doesn't need it; I was referring to the `type='item'`, `name='yarm-selector-tool'`, but it's ok, I'll take care of it in the merge.

**Comment by narc0tiq** [comment]
This being an update to a new version of Factorio, I would grow the version by 0.0.100, so now we get v0.8.201. I'll take care of that in the merge, too, no worries.

Thank you very much for the assist!

**Comment by narc0tiq** [comment]
> The map tags now have icons that look like pins (the new default) rather than having the icon set to the ore type. They still seem to work just fine, but it's a regression in terms of graphics. I also didn't investigate how to fix it.
> 
> Edit: I just looked at this and YARM adds the ore icon to the string text, it doesn't set the tag icon. The default icon simply got more noticeable, so it stands out more than before. No regression, though setting the icon for single ore resources might be a potential improvement.

I went back to test this again and it still looks very bad to me with the ore icon:
image

The problem is that the icon is greatly out of scale with the text, and it looks like they're barely even related. I thought the old circle-style pin was ok, and the new one looks a bit better IMO. I'm ok with it.

**Pull request merged**